src/components/ContactList.tsx

```diff
@@ -37,20 +47,66 @@ const ContactList = () => {
         fetchContacts();
     }, []) 
 
+    //TODO: optimize search so that it can find partial match to name search
+    //TODO: fix the any typing in matchByNameSearResults
+    const getMatchingNames = (nameStr: string | null) => {
+        const namesArray: any[] = [];
+        contacts.map((contact: any) => {
```

Comment:
Why not use `.filter`?

Reply:
Because my approach was iterating over the array of contact objects and storing the value of the name property of the matched contacts in an array. My thought was that I could use filter if I was wanting an array of the objects that matched versus their name property values. That being said, there ought to be a way for me to filter the objects and then use that to render the contact list.